.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,62 @@
+name: test
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  ubuntu-build-test:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: src/fastdds_python
+
+      - name: Install apt dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y \
+            python3 \
+            python3-pip \
+            libasio-dev \
+            libtinyxml2-dev \
+            swig \
+            libpython3-dev
+
+      - name: Install python dependencies
+        run: |
+          sudo pip3 install -U \
+            vcstool \
+            colcon-common-extensions
+
+      - name: Fetch repositories
+        run: |
+          vcs import --skip-existing src < src/fastdds_python/fastdds_python.repos
+          cd src/fastdds
+          git checkout feature/good-datawriter-api
+          cd ../../
```

Comment:
We should create a follow-up PR removing these lines, to be merged after eProsima/Fast-DDS#2560